langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/OracleEmbeddingStore.java

```diff
@@ -641,16 +605,13 @@ public Builder embeddingTable(EmbeddingTable embeddingTable) {
         }
 
         /**
-         * Configures the creation of an index on the embedding column of the {@link EmbeddingTable} used by the
-         * embedding store. Depending on which CreateOption is provided, an index may be created when {@link #build()}
-         * is called. The default createOption is {@link CreateOption#DO_NOT_CREATE}.
-         *
-         * @param createOption Option for creating the index. Not null.
-         * @return This builder. Not null.
+         * Configures the indexes that will be created on the {@link EmbeddingTable}. Two types of
+         * indexes can be created {@link IVFIndex} and {@link JSONIndex}.
+         * @param indexBuilders Indexes to create.
+         * @return This builder.
          */
-        public Builder vectorIndex(CreateOption createOption) {
-            ensureNotNull(createOption, "createOption");
-            vectorIndexCreateOption = createOption;
+        public Builder index(IndexBuilder... indexBuilders) {
```

Comment:
Wondering why the argument type is `IndexBuilder` and not `Index`

Reply:
So that the build method can be called while constructing the store and pass the embedding table as parameter. Another option would be to leave placeholders on the create and drop statements and fill them when the store is being created. I will see what I can do.